tests/components/recorder/test_websocket_api.py

```diff
@@ -794,6 +794,217 @@ def next_id():
+    expect = 5
+    # FIXME: this returns None, I might have expected 5 given the behavior of other 5 minute intervals
+    # await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # Five minutes of growth in STS-only, with a minute offset. Despite that this does not cover the full period, result is still 5
+    start_time = start.replace(hour=6, minute=16)
+    end_time = start.replace(hour=6, minute=21)
+    expect = 5
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # 7 minutes of growth in STS-only, spanning two intervals
+    start_time = start.replace(hour=6, minute=14)
+    end_time = start.replace(hour=6, minute=21)
+    expect = 10
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # One hours worth of growth in LTS-only, with arbitrary minute offsets
+    # Since this does not fully cover the hour, result is None?
+    start_time = start.replace(hour=1, minute=40)
+    end_time = start.replace(hour=2, minute=12)
+    expect = None
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # One hours worth of growth in LTS-only, with arbitrary minute offsets, covering a whole 1-hour period
+    start_time = start.replace(hour=1, minute=40)
+    end_time = start.replace(hour=3, minute=12)
+    expect = 60
+    # Pre-115291 says "-65"
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # 90 minutes of growth in window overlapping LTS+STS/STS-only (4:41 - 6:11)
+    start_time = start.replace(hour=4, minute=41)
+    end_time = start_time + timedelta(minutes=90)
+    expect = 90
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # 4 hours of growth in overlapping LTS-only/LTS+STS (2:01-6:01)
+    start_time = start.replace(hour=2, minute=1)
+    end_time = start_time + timedelta(minutes=240)
+    expect = 185  # 60 from LTS (3:00-3:59), 125 from STS (25 intervals) (4:00-6:01)
+    # Pre-115291 says "120", which is too little
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # 4 hours of growth in overlapping LTS-only/LTS+STS (1:31-5:31)
+    start_time = start.replace(hour=1, minute=31)
+    end_time = start_time + timedelta(minutes=240)
+    expect = 215  # 120 from LTS (2:00-3:59), 95 from STS (19 intervals) 4:00-5:31
+    # Pre-115291 says "90", which is only the STS window (4:00-5:31) ?
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # 5 hours of growth, start time only (1:31-end)
+    start_time = start.replace(hour=1, minute=31)
+    end_time = None
+    # will be actually 2:00 - end
+    expect = 4 * 60 + 30
+    # Pre-115291 says "115" (1 hour and 55 minutes)? which I do not understand
+    # FIXME: Postfix says 240, which doesn't seem right either, I would have expected 270 or 275? for 2:00 to the end
+    # await assert_change_during_fixed(client, start_time, end_time, expect)
```

Comment:
I'm confused here, the total at 2:00 is 120, and the total at the end is 360 so getting 240 back seems reasonable?

Reply:
360 is the value associated with 6:00, but I kept inserting STS until 6:30, so I thought that the end value would be 390.

Reply:
OK, but this passes:
```diff
diff --git a/tests/components/recorder/test_websocket_api.py b/tests/components/recorder/test_websocket_api.py
index a83080a588d..3a1e6effe9d 100644
--- a/tests/components/recorder/test_websocket_api.py
+++ b/tests/components/recorder/test_websocket_api.py
@@ -831,6 +831,8 @@ async def test_statistic_during_period_partial_overlap(
         for i in range(36)
     ]

+    assert imported_stats_5min[-1]["sum"] - imported_stats_hours[0]["sum"] == 360
+
     statId = "sensor.test_overlapping"
     imported_metadata = {
         "has_mean": False,
```

Reply:
Actually seems like I was inserting even more than I thought, I meant to insert STS until 6:30 but I did my count wrong and actually kept inserting them until 7:00, so the end value in STS was actually 420. 

I've corrected this but it broke another testcase.

Reply:
If you want the 5-minute stats not to end at the full hour, I think this needs to change to not inject a multiple of 12 readings:

```py
    # The datapoint at i=0 (start = 4:00) would be the sensor's value at t=4:05, or 245
    imported_stats_5min = [
        {
            "start": (start + timedelta(hours=4, minutes=5 * i)),
            "sum": 4 * 60 + (i + 1) * 5,
        }
        for i in range(36)
    ]
```